lasagne/layers/helper.py

```diff
@@ -184,6 +190,24 @@ def get_output(layer_or_layers, inputs=None, **kwargs):
                                  "mapping this layer to an input expression."
                                  % layer)
             all_outputs[layer] = layer.get_output_for(layer_inputs, **kwargs)
+            if isinstance(layer.get_output_for, MethodType):
+                # Mock in python2 does not support introspection
```

Comment:
`layer.get_output_for` could also be a function if somebody monkey-patches an instance. This won't be a common use case, but I'm still a bit worried about making the implementation less generic just to cater for deficits in the test framework.
To make it a little more robust, what about either:
```python
if isinstance(layer.get_output_for, (MethodType, FunctionType)):
```
or:
```python
try:
    names, _, _, defaults = getargspec(layer.get_output_for)
except TypeError:
    pass  # if introspection is not possible, skip it
else:
    if defaults is not None:
        accepted_kwargs |= set(names[-len(defaults):])
```